x-pack/packages/kbn-elastic-assistant/impl/connectorland/connector_selector/index.tsx

```diff
@@ -140,7 +148,7 @@ export const ConnectorSelector: React.FC<Props> = React.memo(
 
         const apiProvider = (
           connectors?.find((c) => c.id === connectorId) as ActionConnectorProps<Config, unknown>
-        )?.config.apiProvider as OpenAiProviderType;
+        )?.config?.apiProvider as OpenAiProviderType;
```

Comment:
Going forward, lets validate these objects so we can have better branching based on what they are (preconfigured or not w/o config).

Reply:
Extracted to common function for easier cleanup in [2514540](https://github.com/elastic/kibana/pull/164900/commits/2514540b10c7878af1b4a64c0dd20e286a7e1352)